### PR TITLE
Changing builds to JRE, and moving Java8 to pinned version for forge

### PR DESCRIPTION
**Pull request by shotah**
Changes:
* Added JDK 17 image
* Changed base images to JRE
* Added curse forge test
* quote fix on forge api

Issues:
* https://github.com/itzg/docker-minecraft-server/issues/1380
* https://github.com/itzg/docker-minecraft-server/issues/1381

### CODE REVIEW

**Comment by shotah** [comment]
@itzg Let me know when you get a chance to review this!

File: .github/workflows/build-multiarch.yml [hunk 1]
```diff
@@ -27,35 +27,40 @@ jobs:
           - java11-openj9
         include:
           - variant: java17
+            baseImage: eclipse-temurin:17-jre
+            platforms: linux/amd64,linux/arm/v7,linux/arm64
+            mcVersion: LATEST
+          - variant: java17jdk
```
Comment (itzg) [inline .github/workflows/build-multiarch.yml:33]: These need to be dash delimited, like
```suggestion
          - variant: java17-jdk
```

File: .github/workflows/build-multiarch.yml [hunk 2]
```diff
@@ -27,35 +27,40 @@ jobs:
...
-            baseImage: ibm-semeru-runtimes:open-17-jdk
+            baseImage: ibm-semeru-runtimes:open-17-jre
             platforms: linux/amd64,linux/arm64
             mcVersion: LATEST
           - variant: java17-alpine
-            baseImage: eclipse-temurin:17-jdk-alpine
+            baseImage: eclipse-temurin:17-jre-alpine
             platforms: linux/amd64
             mcVersion: LATEST
           - variant: java8
```
Comment (itzg) [inline .github/workflows/build-multiarch.yml:45]: There also needed to be a `java8-jdk` variant added as a derivative of java8-multiarch.

**Review by itzg — CHANGES_REQUESTED** [review]
I'm going to go ahead and make the java8-multiarch change because apparently it was super urgent.

File: .github/workflows/build-multiarch.yml [hunk 3]
```diff
@@ -27,35 +27,40 @@ jobs:
...
             mcVersion: 1.12.2
           - variant: java8-multiarch
-            baseImage: eclipse-temurin:8-jdk
+            # NOTE: Unable to go past 8u312 because of Forge https://github.com/MultiMC/Launcher/issues/4470
+            baseImage: eclipse-temurin:8u312-b07-jre
             platforms: linux/amd64,linux/arm64
             mcVersion: 1.12.2
           - variant: java8-openj9
-            baseImage: ibm-semeru-runtimes:open-8-jdk
+            baseImage: ibm-semeru-runtimes:open-8-jre
```
Comment (itzg) [inline .github/workflows/build-multiarch.yml:55]: and I guess this one need to be pinned too

**Comment by itzg** [comment]
Pinning is done in https://github.com/itzg/docker-minecraft-server/pull/1383

**Comment by itzg** [comment]
I fixed the forge API key and then test failures showed up:

https://github.com/itzg/docker-minecraft-server/actions/runs/1893247260

**Comment by shotah** [comment]
@itzg thanks for figuring out why that key wasn't working. I've figured out the test failures, the mods are built for those versions of minecraft. I've added minecraft version check into require.sh file. The mods probably need to be pinned at 1.18, but I figure we can cross that bridge once the dust has settled. 

Let me know if you see anything else that needs to be updated.

**Review by itzg — CHANGES_REQUESTED** [review]
One last tweak

**Comment by shotah** [comment]
@itzg Let me know what you think.

**Review by itzg — APPROVED** [review]

**Pull request merged**